src/main/java/com/aws/greengrass/lifecyclemanager/GenericExternalService.java

```diff
@@ -584,7 +613,8 @@ public void handleError() throws InterruptedException {
                 Lifecycle.TIMEOUT_NAMESPACE_TOPIC));
 
         CountDownLatch handlerExecutionCdl = new CountDownLatch(1);
-        run(Lifecycle.LIFECYCLE_RECOVER_NAMESPACE_TOPIC, c -> handlerExecutionCdl.countDown(), lifecycleProcesses);
+        run(Lifecycle.LIFECYCLE_RECOVER_NAMESPACE_TOPIC, c -> handlerExecutionCdl.countDown(), lifecycleProcesses,
```

Comment:
nit - would've been nice to leave the original run method and have it call the new method with `runImmediately` as `true` so that you don't see booleans passed around in a lot of places